src/fparser/readfortran.py

```diff
@@ -810,23 +810,30 @@ def format_message(self, kind, message, startlineno, endlineno,
                 else:
                     l1 = sourceline[startcolno:endcolno]
                     l2 = sourceline[endcolno:]
-                r.append('%s%s%s <== %s' % (l0,yellow_text(l1),l2,red_text(message)))
+                r.append('%s%s%s <== %s' % (l0, yellow_text(l1), l2,
+                                            red_text(message)))
             else:
-                r.append(linenostr+ self.source_lines[i-1])
+                r.append(linenostr + self.source_lines[i-1])
         return '\n'.join(r)
 
     def format_error_message(self, message, startlineno, endlineno,
                              startcolno=0, endcolno=-1):
-        return self.format_message('ERROR',message, startlineno,
+        '''Create a string with an error message.'''
+        return self.format_message('ERROR', message, startlineno,
                                    endlineno, startcolno, endcolno)
 
-    def format_warning_message(self, message, startlineno, endlineno,
-                               startcolno=0, endcolno=-1):
-        return self.format_message('WARNING',message, startlineno,
+    def format_warning_message(self, message, startlineno=None,
+                               endlineno=None, startcolno=0, endcolno=-1):
+        '''Create a string with a warning message.'''
+        if startlineno is None:
```

Comment:
Please add a comment to explain why/how you adjust start/endlineno.

Reply:
In fact, these lines aren't covered by the test suite either. Please can you add/modify tests such that they are.

Reply:
Thanks for adding the docstring. However, the new lines of code are not covered. Also, what happens if `startlineno=None` and `self.linecount < 2`?

Reply:
This confuses me a bit ... while I agree that in my original pull request I had not handled that case, but I've fixed that, see: https://github.com/stfc/fparser/pull/55/files:
 +      if startlineno is None:
+            startlineno = self.linecount-2
+            if startlineno < 0:
+                startlineno = 1

The change above is marked as 'outdated' - I assume there was a mis-click somewhere ;)